### PR TITLE
Do not allow login or register over an insecure connection

### LINKED ISSUE #1149: New Login page lets me log in via port 80
I just noticed that the new login page lets me login via an unsecured connection. 

I can access and login via port 80 and only after the login, it displays the "Your access to BTCPay Server is over an unsecured network..." message. Even though the message displays, I'm logged in and have access to all functions.

### PR DESCRIPTION
**Pull request by Kukks**
fixes #1149

### CODE REVIEW

**Comment by NicolasDorier** [comment]
Can you rebase and run the tests again?

File: BTCPayServer/Controllers/AccountController.cs [hunk 1]
```diff
@@ -580,6 +619,21 @@ private IActionResult RedirectToLocal(string returnUrl = null)
                 return RedirectToAction(nameof(HomeController.Index), "Home");
             }
         }
+        
+        
+        private bool CanLoginOrRegister()
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/AccountController.cs]: Can you refactor this? This is not obvious that this method has side effects. I was wondering why you were calling it without checking the result.

Comment (Kukks) [reply BTCPayServer/Controllers/AccountController.cs]: What do you want me to do? Rename it or split into two

Comment (NicolasDorier) [reply BTCPayServer/Controllers/AccountController.cs]: Split in two

**Pull request merged**